pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,18 +98,89 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
 	return nil
 }
 
+func checkZoneIsMaster(context *Context, realmArg string, zoneGroupArg string, zoneArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	masterZoneID, err := decodeMasterZoneID(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == masterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getZoneEndpoints(context *Context, realmArg string, zoneGroupArg string, zoneName string, serviceEndpoint string) (string, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zones, err := decodeZonesList(zoneGroupOutput)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse zones list")
+	}
+
+	numZones := len(zones)
+	zoneEndpoints := ""
+	for i := 0; i < numZones; i++ {
+		if zones[i].Name == zoneName {
+			numEndpoints := len(zones[i].Endpoints)
+			for j := 0; j < numEndpoints; j++ {
+				// in case object-store operator code is rereconciled, zone modify could get run again with serviceEndpoint added again
+				if zones[i].Endpoints[j] != serviceEndpoint {
+					zoneEndpoints += zones[i].Endpoints[j] + ","
```

Comment:
I'd suggest appending to a `[]string`, then the caller of the method can combine them to a single string with `strings.Join(endpoints, ",")`